Forcefully disable RVV code that doesn't build because of a long
vs long long mismatch.

Index: numpy/_core/src/highway/hwy/detect_targets.h
--- numpy/_core/src/highway/hwy/detect_targets.h.orig
+++ numpy/_core/src/highway/hwy/detect_targets.h
@@ -334,7 +334,8 @@
 // HWY_RVV fails to compile with GCC < 13 or Clang < 16.
 #if HWY_ARCH_RISCV &&                                     \
     ((HWY_COMPILER_CLANG && HWY_COMPILER_CLANG < 1600) || \
-     (HWY_COMPILER_GCC_ACTUAL && HWY_COMPILER_GCC_ACTUAL < 1300))
+     (HWY_COMPILER_GCC_ACTUAL && HWY_COMPILER_GCC_ACTUAL < 1300) || \
+     defined(__OpenBSD__))
 #define HWY_BROKEN_RVV (HWY_RVV)
 #else
 #define HWY_BROKEN_RVV 0
